napari/utils/translations.py

```diff
@@ -0,0 +1,418 @@
+    )
+    loc = babel.Locale.parse(locale)
+    return loc.get_display_name(display_locale).capitalize()
+
+
+def _is_valid_locale(locale: str) -> bool:
+    """
+    Check if a `locale` value is valid.
+
+    Parameters
+    ----------
+    locale: str
+        Language locale code.
+
+    Notes
+    -----
+    A valid locale is in the form language (See ISO-639 standard) and an
+    optional territory (See ISO-3166 standard).
+
+    Examples of valid locales:
+    - English: DEFAULT_LOCALE
+    - Australian English: "en_AU"
+    - Portuguese: "pt"
+    - Brazilian Portuguese: "pt_BR"
+
+    Examples of invalid locales:
+    - Australian Spanish: "es_AU"
+    - Brazilian German: "de_BR"
+    """
+    # This is a dependency of the language packs to keep out of core
+    import babel
+
+    valid = False
+    try:
+        babel.Locale.parse(locale)
+        valid = True
+    except babel.core.UnknownLocaleError:
+        pass
+    except ValueError:
+        pass
+
+    return valid
+
+
+def get_language_packs(display_locale: str = DEFAULT_LOCALE) -> dict:
+    """
+    Return the available language packs installed in the system.
+
+    The returned information contains the languages displayed in the current
+    locale. This can be used to generate the preferences dialog information.
+
+    Parameters
+    ----------
+    display_locale: str, optional
+        Default is DEFAULT_LOCALE.
+
+    Returns
+    -------
+    dict
+        A dict with the native and display language for all locales found.
+
+    Example
+    -------
+    >>> get_language_packs("es_CO")
+    {
+        'en': {'displayName': 'Inglés', 'nativeName': 'English'},
+        'es_CO': {
+            'displayName': 'Español (colombia)',
+            'nativeName': 'Español (colombia)',
+        },
+    }
+    """
+    from napari_plugin_engine.manager import iter_available_plugins
+
+    lang_packs = iter_available_plugins(NAPARI_LANGUAGEPACK_ENTRY)
+    found_locales = {k: v for (k, v, _) in lang_packs}
+
+    invalid_locales = []
+    valid_locales = []
+    messages = []
+    for locale in found_locales:
+        if _is_valid_locale(locale):
+            valid_locales.append(locale)
+        else:
+            invalid_locales.append(locale)
+
+    display_locale = (
+        display_locale if display_locale in valid_locales else DEFAULT_LOCALE
+    )
+    locales = {
+        DEFAULT_LOCALE: {
+            "displayName": _get_display_name(DEFAULT_LOCALE, display_locale),
+            "nativeName": _get_display_name(DEFAULT_LOCALE, DEFAULT_LOCALE),
+        }
+    }
+    for locale in valid_locales:
+        locales[locale] = {
+            "displayName": _get_display_name(locale, display_locale),
+            "nativeName": _get_display_name(locale, locale),
+        }
+
+    if invalid_locales:
+        messages.append(
+            f"The following locales are invalid: {invalid_locales}!"
+        )
+
+    return locales
```

Comment:
@tlambert03 this was the reason for the display name func. This will be used when generating the preferences dialog.